pandas/core/generic.py

```diff
@@ -5601,53 +5601,83 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
         """
         Trim values at input threshold(s).
 
+        Elements above/below the upper/lower thresholds will be changed to
+        upper/lower thresholds.
+
         Parameters
         ----------
-        lower : float or array_like, default None
-        upper : float or array_like, default None
-        axis : int or string axis name, optional
-            Align object with lower and upper along the given axis.
+        lower : float, array-like or None, default None
+            Lower threshold for clipping.  Values smaller than `lower` will be
+            converted to `lower`.
+        upper : float, array-like or None, default None
+            Upper threshold for clipping.  Values larger than `upper` will be
+            converted to `upper`.
+        axis : {0 or 'index', 1 or 'columns', None}, default None
+            Apply clip by index (i.e. by rows) or columns.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        *args : Additional keywords have no effect but might be accepted
+            for compatibility with numpy.
+        **kwargs : Additional keywords have no effect but might be accepted
+            for compatibility with numpy.
 
         Returns
         -------
-        clipped : Series
+        `Series` or `DataFrame`.
+            DataFrame is returned with those values above/below the
+            `upper`/`'lower` thresholds set to the threshold values.
+
+        See Also
+        --------
+        pandas.Series.clip : Trim values at input threshold(s).
 
         Examples
         --------
+        >>> some_data = {'a': [1, 2, 3], 'b': [4, 5, 6], 'c': [7, 8, 9001]}
+        >>> df = pd.DataFrame(some_data, index = ['foo', 'bar', 'foobar'])
```

Comment:
The spaces around `=` are not PEP-8 compliant. Sorry the validation script doesn't check it yet.